integration/spark/src/main/scala/org/apache/carbondata/spark/CarbonOption.scala

```diff
@@ -84,5 +84,22 @@ class CarbonOption(options: Map[String, String]) {
 
   lazy val dateformat: Option[String] = options.get("dateformat")
 
+  lazy val SPATIAL_INDEX: Option[String] = options.get("SPATIAL_INDEX")
+
+  lazy val SPATIAL_INDEX_type: Option[String] = options.get("SPATIAL_INDEX.mygeohash.type")
```

Comment:
looks like the newly added options are case sensitive. please handle with case insensitive options

Reply:
the options map is already normalized. so, case insensitive values are also accepted. Edited existing testcase with different cases